File: Python/Project_Euler_S7.py
# Project Euler – Problem 7
# Title: 10001st Prime
# Link: https://projecteuler.net/problem=7
#
# Find the nth prime number.

def is_prime(a):
    if a <= 1:
        return False
    
    elif a == 2:
        return True

    elif a % 2 == 0:
        return False
    
    for i in range(3, int(a**0.5) + 2):
        if a % i == 0:
            return False
        
    return True

primes = [2]
x = 3
while len(primes) <= 10000:
    if is_prime(x):
        primes.append(x)
    x += 2

print(len(primes),"st prime" , primes[-1])

File: C/Project_Euler_S7.c
// Project Euler – Problem 7
// Title: 10001st Prime
// Link: https://projecteuler.net/problem=7
//
// Find the nth prime number.

#include <stdio.h>

bool is_prime(int a)
{
    if (a <= 1)
    {    
        return false;
    }
    else if (a == 2)
    {    
        return true;
    }
    else if (a % 2 == 0)
    {  
        return false;
    }

    for (int i = 3; i * i <= a + 2; i += 2)
        {
            if (a % i == 0)
            {
                return false;
            }
        }
    return true;
}

int main()
{

long long count = 0; 
int x  = 0;
while (count <= 10000)
{
    x++;
    if (is_prime(x))
    {
        count++;
    }
}

printf("10001st prime: %d" , x);
}